include/bluetooth/services/ots_srv.h

```diff
@@ -0,0 +1,644 @@
+	 *              NULL if object is created by the following function:
+	 *              bt_gatt_ots_srv_object_add().
+	 *  @param id   Object ID.
+	 *  @param init Object initialization metadata.
+	 *
+	 *  @return 0 in case of success or negative value in case of error.
+	 *  Possible return values:
+	 *  -ENOMEM if no available space for new object.
+	 */
+	int (*obj_created)(struct bt_gatt_ots_srv *ots, struct bt_conn *conn,
+			   uint64_t id,
+			   const struct bt_gatt_ots_srv_obj_metadata *init);
+
+	/** Object deleted callback
+	 *
+	 *  This callback is called whenever an object is deleted. It is
+	 *  also triggered when the server deletes an object with
+	 *  bt_gatt_ots_srv_object_delete() API.
+	 *
+	 *  @param ots  OTS Server instance.
+	 *  @param conn The connection that deleted the object or NULL if
+	 *              this request came from the server.
+	 *  @param id   Object ID.
+	 */
+	void (*obj_deleted)(struct bt_gatt_ots_srv *ots, struct bt_conn *conn,
+			    uint64_t id);
+
+	/** Object selected callback
+	 *
+	 *  This callback is called on successful object selection.
+	 *
+	 *  @param ots  OTS Server instance.
+	 *  @param conn The connection that selected new object.
+	 *  @param id   Object ID.
+	 */
+	void (*obj_selected)(struct bt_gatt_ots_srv *ots, struct bt_conn *conn,
+			     uint64_t id);
+
+	/** Object read callback
+	 *
+	 *  This callback is called multiple times during the Object read
+	 *  operation. OTS module will keep requesting successive Object
+	 *  fragments from the application until the read operation is
+	 *  completed. The end of read operation is indicated by NULL data
+	 *  parameter.
+	 *
+	 *  @param ots    OTS Server instance.
+	 *  @param conn   The connection that read object.
+	 *  @param id     Object ID.
+	 *  @param data   In:  NULL once the read operations is completed.
+	 *                Out: Next chunk of data to be sent.
+	 *  @param len    Remaining length requested by the client.
+	 *  @param offset Object data offset.
+	 *
+	 *  @return Data length to be sent via data parameter. This value
+	 *          shall be smaller or equal to the len parameter.
+	 */
+	uint32_t (*obj_read)(struct bt_gatt_ots_srv *ots, struct bt_conn *conn,
+			     uint64_t id, uint8_t **data, uint32_t len,
+			     uint32_t offset);
+};
+
+/** @brief Descriptor for OTS initialization. */
+struct bt_gatt_ots_srv_init {
+	/* OTS features */
+	struct bt_gatt_ots_srv_feat features;
+
+	/* Callbacks */
+	struct bt_gatt_ots_srv_cb *cb;
+} __packed;
+
+/** @brief Add an object to the OTS Server.
+ *
+ *  This function adds an object to the OTS database. When the
+ *  object is being added, a callback obj_created() is called
+ *  to notify the user about a new object ID.
+ *
+ *  @param ots      OTS Server instance.
+ *  @param obj_init Meta data of the object.
+ *
+ *  @return 0 in case of success or negative value in case of error.
+ */
+int bt_gatt_ots_srv_object_add(struct bt_gatt_ots_srv *ots,
```

Comment:
For consistency in the API i think it should be `bt_gatt_ots_srv_obj_add`, same for the other places where `_OBJECT_` or `_object_` is used.

Reply:
yes, it will be good to align this